In [1]:
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC, LinearSVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression, LinearRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import GridSearchCV, train_test_split

import pandas as pd
import numpy as np 
import seaborn  as sns
import matplotlib.pyplot as plt

%matplotlib inline 
sns.set('notebook')

import warnings

warnings.filterwarnings("ignore", category=UserWarning)
warnings.filterwarnings("ignore", category=FutureWarning)

In [2]:
ikea=pd.read_csv('IKEA_SA_Furniture_Web_Scrapings_sss.csv')

# STEP 1
Step 1: EDA - shape, head, tail, sample, info

In [3]:
ikea.shape

(3694, 14)

In [10]:
ikea.info()
ikea.dtypes

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3694 entries, 0 to 3693
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         3694 non-null   int64  
 1   item_id            3694 non-null   int64  
 2   name               3694 non-null   object 
 3   category           3694 non-null   object 
 4   price              3694 non-null   float64
 5   old_price          3694 non-null   object 
 6   sellable_online    3694 non-null   bool   
 7   link               3694 non-null   object 
 8   other_colors       3694 non-null   object 
 9   short_description  3694 non-null   object 
 10  designer           3694 non-null   object 
 11  depth              2231 non-null   float64
 12  height             2706 non-null   float64
 13  width              3105 non-null   float64
dtypes: bool(1), float64(4), int64(2), object(7)
memory usage: 378.9+ KB


Unnamed: 0             int64
item_id                int64
name                  object
category              object
price                float64
old_price             object
sellable_online         bool
link                  object
other_colors          object
short_description     object
designer              object
depth                float64
height               float64
width                float64
dtype: object

In [22]:
ikea.isna().sum()
#ikea.notna().sum()

Unnamed: 0              0
item_id                 0
name                    0
category                0
price                   0
old_price               0
sellable_online         0
link                    0
other_colors            0
short_description       0
designer                0
depth                1463
height                988
width                 589
dtype: int64

In [7]:
ikea.columns

Index(['Unnamed: 0', 'item_id', 'name', 'category', 'price', 'old_price',
       'sellable_online', 'link', 'other_colors', 'short_description',
       'designer', 'depth', 'height', 'width'],
      dtype='object')

In [12]:
ikea.describe()

Unnamed: 0       item_id        price        depth       height  \
count  3694.000000  3.694000e+03  3694.000000  2231.000000  2706.000000   
mean   1846.500000  4.863240e+07  1078.208419    54.379202   101.679970   
std    1066.510275  2.888709e+07  1374.652494    29.958351    61.097585   
min       0.000000  5.848700e+04     3.000000     1.000000     1.000000   
25%     923.250000  2.039057e+07   180.900000    38.000000    67.000000   
50%    1846.500000  4.928808e+07   544.700000    47.000000    83.000000   
75%    2769.750000  7.040357e+07  1429.500000    60.000000   124.000000   
max    3693.000000  9.993262e+07  9585.000000   257.000000   700.000000   

             width  
count  3105.000000  
mean    104.470853  
std      71.133771  
min       1.000000  
25%      60.000000  
50%      80.000000  
75%     140.000000  
max     420.000000

In [14]:
ikea.head(20)

Unnamed: 0   item_id                   name       category   price  \
0            0  90420332               FREKVENS  Bar furniture   265.0   
1            1    368814              NORDVIKEN  Bar furniture   995.0   
2            2   9333523  NORDVIKEN / NORDVIKEN  Bar furniture  2095.0   
3            3  80155205                   STIG  Bar furniture    69.0   
4            4  30180504                NORBERG  Bar furniture   225.0   
5            5  10122647                 INGOLF  Bar furniture   345.0   
6            6  70404875               FRANKLIN  Bar furniture   129.0   
7            7  60155602                DALFRED  Bar furniture   195.0   
8            8  50406465               FRANKLIN  Bar furniture   129.0   
9            9  69304221    EKEDALEN / EKEDALEN  Bar furniture  2176.0   
10          10  90404879               FRANKLIN  Bar furniture   149.0   
11          11    121766                 INGOLF  Bar furniture   395.0   
12          12    397736               NORRARYD  Bar furniture   395.0   
13          13  50420329               FREKVENS  Bar furniture   177.0   
14          14    400550               EKEDALEN  Bar furniture   345.0   
15          15  29304826               TOMMARYD  Bar furniture   695.0   
16          16  40426138             HENRIKSDAL  Bar furniture   395.0   
17          17  50363649              KULLABERG  Bar furniture   140.0   
18          18  70246089                JANINGE  Bar furniture   595.0   
19          19  60406785               FRANKLIN  Bar furniture   149.0   

       old_price  sellable_online  \
0   No old price             True   
1   No old price            False   
2   No old price            False   
3   No old price             True   
4   No old price             True   
5   No old price             True   
6   No old price             True   
7   No old price             True   
8   No old price             True   
9       SR 2,375             True   
10  No old price             True   
11  No old price             True   
12  No old price             True   
13        SR 295             True   
14  No old price             True   
15  No old price             True   
16  No old price             True   
17        SR 175             True   
18  No old price             True   
19  No old price             True   

                                                 link other_colors  \
0   https://www.ikea.com/sa/en/p/frekvens-bar-tabl...           No   
1   https://www.ikea.com/sa/en/p/nordviken-bar-tab...           No   
2   https://www.ikea.com/sa/en/p/nordviken-nordvik...           No   
3   https://www.ikea.com/sa/en/p/stig-bar-stool-wi...          Yes   
4   https://www.ikea.com/sa/en/p/norberg-wall-moun...           No   
5   https://www.ikea.com/sa/en/p/ingolf-bar-stool-...           No   
6   https://www.ikea.com/sa/en/p/franklin-bar-stoo...           No   
7   https://www.ikea.com/sa/en/p/dalfred-bar-stool...           No   
8   https://www.ikea.com/sa/en/p/franklin-bar-stoo...           No   
9   https://www.ikea.com/sa/en/p/ekedalen-ekedalen...           No   
10  https://www.ikea.com/sa/en/p/franklin-bar-stoo...           No   
11  https://www.ikea.com/sa/en/p/ingolf-bar-stool-...           No   
12  https://www.ikea.com/sa/en/p/norraryd-bar-stoo...           No   
13  https://www.ikea.com/sa/en/p/frekvens-bar-stoo...           No   
14  https://www.ikea.com/sa/en/p/ekedalen-bar-stoo...           No   
15  https://www.ikea.com/sa/en/p/tommaryd-table-an...           No   
16  https://www.ikea.com/sa/en/p/henriksdal-bar-st...           No   
17  https://www.ikea.com/sa/en/p/kullaberg-stool-b...          Yes   
18  https://www.ikea.com/sa/en/p/janinge-bar-stool...           No   
19  https://www.ikea.com/sa/en/p/franklin-bar-stoo...           No   

                                    short_description  \
0            Bar table, in/outdoor,          51x51 cm   
1                       Bar table,          140x80 cm   
2                          Ba

In [17]:
ikea.duplicated().sum()

0

# Conclusions on STEP 1.
Dataset Ikea consists 3694 rows and 14 columns with following datatypes: bool(1), float64(4), int64(2), object(7). 
The majority of columns have no null values, except the last three ones: depth (1463 Null values), height (988), width (589). 
No duplicated values were found in the dataset. 
Due to this primar research it is obvious that the main problem is the presence of null values in the three columns which define the product dimensions - depth, height and width. The number of null values are far less then 50%, so in this case I wouldn't suggest dropping this columns. As an alternative, we can fill the data with mean or mode values. Also we can use the Nearest Neighbour method to predict the missing numbers in regard to the closest ones.